lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -597,26 +634,58 @@ public static function get_merged_data( $origin = 'custom' ) {
 			_deprecated_argument( __FUNCTION__, '5.9.0' );
 		}
 
+		if ( ! in_array( $origin, WP_Theme_JSON_Gutenberg::VALID_ORIGINS, true ) ) {
+			$origin = 'custom';
+		}
+
+		// Map origins to block cache values.
+		$cache_map = array(
```

Comment:
What if we update the `$blocks_cache` to have the proper keys (`default` instead of `core`, and `custom` instead of `user`) so we don't need this?

Reply:
@oandregal While I agree this would be ideal, I think this should be subject to another separate issue/PR. There may be backward compatibility implications from changing the cache names - probably not, but that would be a fix not really related to what this PR is aiming to achieve.

Reply:
If there's a path to making these consistent, I'm all for it, but was out of scope of what I was trying to show here, so I just worked with what existed.